platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/generic/MetadataRendererRegistry.java

```diff
@@ -123,16 +151,34 @@ public SimpleHtmlWriter render(Object metadata, SimpleHtmlWriter htmlWriter) thr
         }
     }
 
-    public static final class URIMetadataRenderer implements MetadataRenderer<URI> {
+    public static final class ClickableLinkRenderer implements MetadataRenderer {
         @Override
-        public Class<URI> getMetadataType() {
-            return URI.class;
+        public Set<Class<?>> getMetadataTypes() {
+            return new HashSet<>(Arrays.asList(URI.class, File.class, RegularFile.class));
         }
 
         @Override
         public SimpleHtmlWriter render(Object metadata, SimpleHtmlWriter htmlWriter) throws IOException {
-            return (SimpleHtmlWriter) htmlWriter.startElement("a").attribute("href", metadata.toString())
-                .characters(MetadataRenderer.trimIfNecessary(metadata.toString()))
+            String text;
+            String link;
+            if (metadata instanceof File) {
+                link = TextUtil.normaliseFileSeparators(((File) metadata).toURI().toString());
```

Comment:
I don't think we need to normalize file separators here because they're always a URL, so this can just be `...toURI().toASCIIString()`

I think this applies for all cases because the link should not contain non-ASCII characters, but the human readable part can.